src/Symfony/Component/Security/Http/Authentication/AuthenticationUtils.php

```diff
@@ -31,11 +31,9 @@ public function __construct(RequestStack $requestStack)
     }
 
     /**
-     * @param bool $clearSession
-     *
      * @return AuthenticationException|null
      */
-    public function getLastAuthenticationError($clearSession = true)
+    public function getLastAuthenticationError(bool $clearSession = true)
```

Comment:
This could receive an `?AuthenticationException` return type and the doc can be removed after that. Not sure if that's within the scope of your PR though. If this is within the scope of your changes, there are a bunch of other return types that could be added as well on other functions that would reduce the need of docblocks.

The name of this variable doesn't make much sense, you don't clear the session, only remove the authentication error. My proposal is to change this to `$removeFromSession` or `$clearFromSession`, if not in this PR, in a new one :smile: 

The last point is not related to your changes, but perhaps possible to take along? (/cc @chalasr).

Reply:
Return types are out of scope for this PR, see #32179. Also, adding one would be a breaking change.